sdk/common/src/test/scala/org/typelevel/otel4s/sdk/internal/ComponentRegistrySuite.scala

```diff
@@ -0,0 +1,77 @@
+/*
+ * Copyright 2023 Typelevel
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.typelevel.otel4s.sdk.internal
+
+import cats.effect.IO
+import munit.CatsEffectSuite
+import org.typelevel.otel4s.Attribute
+import org.typelevel.otel4s.sdk.Attributes
+
+class ComponentRegistrySuite extends CatsEffectSuite {
+
+  private val name = "component"
+  private val version = "0.0.1"
+  private val schemaUrl = "https://otel4s.schema.com"
+  private val attributes = Attributes(Attribute("key", "value"))
+
+  registryTest("get cached values (by name only)") { registry =>
+    for {
+      v1 <- registry.get(name, None, None, Attributes.Empty)
+      v2 <- registry.get(name, None, None, attributes)
+      v3 <- registry.get(name, Some(version), None, attributes)
+      v4 <- registry.get(name, Some(version), Some(schemaUrl), attributes)
```

Comment:
missing a case in each of the three tests for name and schema only, as well as a fourth test for name and schema only